docs/api/basic/useSnapshot.mdx

```diff
@@ -7,12 +7,17 @@ description: 'Create a local snapshot that catches changes.'
 
 # useSnapshot
 
-Create a local snapshot that catches changes. This hook actually returns a wrapped snapshot in a proxy for
-render optimization instead of a plain object, which is what <a href="/docs/api/advanced/snapshot">`snapshot()`</a> returns.
+Create a local snapshot that catches changes.
 
-#### Rule of thumb: read from snapshots in render function, otherwise use the source.
+Normally, Valtio's snapshots (created via <a href="/docs/api/advanced/snapshot">`snapshot()`</a>) are recreated on _any_ change to a store, or any of its child stores.
 
-The component will only re-render when the parts of the state you access have changed; it is render-optimized.
+However `useSnapshot` wraps the Valtio snapshot in an access-tracking proxy, so your component is render optimized, i.e. it will only re-render if keys that it (or its child components) specifically accessed have changed, and not on every single change to the store.
+
+## Usage
+
+### Read from snapshots in render, write to the store in callbacks
+
+Snapshots are read-only, so you can render JSX from them, but mutations still need to be made via the store.
```

Comment:
we want to emphasize, when reading in a function (like `handleClick`), we should read it from the proxy state.

btw, I don't think we (should) use the term `store` in docs.

Reply:
Good point, I reworded + gave example of the callback using the proxy state.

> use the term store in docs.

Wdyt of the term state?

My hesitation about the term proxy is, ironically, the same hesitation you had wrt getters: that it seems like an advanced term/concept most users probably won't know about / shouldn't have to know about, as ideally/imo that Valtio uses proxies is just an implementation detail.

Vs. "store", which I think is generally widely used in state management circles for "the store of your state". So it's a more "logical"/concept based term than proxy's physical term.

...I would even pitch that in some Valtio 2.x, maybe `proxy` is renamed to `store`. :-D

("state" I think is too ambigious...do you mean my store state, or my snapshot state? Both the proxy and snapshot have state, it's just that one is mutable state and other immutable state. Hence like the store (mutable) + snapshot (immutable) pair.) 

That said, don't want that terminology suggestion to hold up the PR, so I've replaced all the "store"s with "proxy", unless you find one I've missed.

Reply:
I use "state" for valtio proxy objects and "snap" or "snapshot" for valtio snapshot objects.
I know "state" can be confusing and too general, though. We can use "proxy state" to be a little more explicit.